runner/src/main.rs

```diff
@@ -290,64 +291,107 @@ fn build_server(opt: &BuildServer) -> Step {
+                    ServerVariant::Base => "",
+                    ServerVariant::NoIntrospectionClient => "oak-unsafe",
+                    ServerVariant::Unsafe => "oak-unsafe,oak-introspection-client",
+                    // If building in coverage mode, use the default target from the host, and build
+                    // in unsafe (debug) mode.
+                    ServerVariant::Coverage => "oak-unsafe,oak-introspection-client",
+                    ServerVariant::Experimental => "oak-attestation,awskms,gcpkms,oak-unsafe,oak-introspection-client",
+                }, &if opt.server_variant == ServerVariant::Coverage {
+                    hashmap! {
+                        // Build the Runtime server in coverage mode, as per https://github.com/mozilla/grcov
+                        "CARGO_INCREMENTAL".to_string() => "0".to_string(),
+                        "RUSTDOCFLAGS".to_string() => "-Cpanic=abort".to_string(),
+                        // grcov instructions suggest also including `-Cpanic=abort` in RUSTFLAGS, but this causes our build.rs scripts to fail.
+                        "RUSTFLAGS".to_string() => "-Zprofile -Ccodegen-units=1 -Copt-level=0 -Clink-dead-code -Coverflow-checks=off -Zpanic-abort_tests".to_string(),
+                    }
+                } else {
+                    hashmap! {}
+                },)
+            ],
         ].into_iter()
             .flatten()
             .collect::<Vec<_>>()
     }
 }
 
+fn build_functions_server(opt: &BuildFunctionsServer) -> Step {
+    Step::Multiple {
+        name: "server".to_string(),
+        steps: vec![
+            vec![Step::Single {
+                name: "create bin folder".to_string(),
+                command: Cmd::new(
+                    "mkdir",
+                    vec!["-p".to_string(), "oak_functions/loader/bin".to_string()],
+                ),
+            }],
+            vec![match opt.server_variant {
+                FunctionsServerVariant::Unsafe => build_server_step(
+                    "oak_functions/loader",
+                    &opt.server_rust_toolchain,
+                    opt.server_rust_target
+                        .as_deref()
+                        .unwrap_or(DEFAULT_SERVER_RUST_TARGET),
+                    "oak-unsafe",
+                    &hashmap! {},
+                ),
+                FunctionsServerVariant::Base => build_server_step(
+                    "oak_functions/loader",
+                    &opt.server_rust_toolchain,
+                    opt.server_rust_target
+                        .as_deref()
+                        .unwrap_or(DEFAULT_SERVER_RUST_TARGET),
+                    "",
+                    &hashmap! {},
+                ),
+            }],
+        ]
+        .into_iter()
+        .flatten()
+        .collect::<Vec<_>>(),
+    }
+}
+
+fn build_server_step(
```

Comment:
This is basically a `build_rust_binary` function, right?

Reply:
Done.